plover/app.py

```diff
@@ -105,6 +107,13 @@ def update_engine(engine, old, new):
     if old.get_space_placement() != space_placement:
         engine.set_space_placement(space_placement)
 
+    start_capitalized = new.get_start_capitalized()
+    start_attached = new.get_start_attached()
+    if (old.get_start_capitalized() != start_capitalized or
+            old.get_start_attached() != start_attached):
+        engine.set_starting_stroke_state(attach=start_attached,
+                                         capitalize=start_capitalized)
+
 def same_thread_hook(fn, *args):
```

Comment:
We should really factorize `init_engine/reset_machine/update_engine` to avoid duplicated code for each parameter, maybe something along those lines:

```diff
diff --git i/plover/app.py w/plover/app.py
index 7edfd96..d5aab2f 100644
--- i/plover/app.py
+++ w/plover/app.py
@@ -39,43 +39,31 @@ class SimpleNamespace(object):

 def init_engine(engine, config):
     """Initialize a StenoEngine from a config object."""
-    reset_machine(engine, config)
-
-    dictionary_file_names = config.get_dictionary_file_names()
-    try:
-        dicts = dict_manager.load(dictionary_file_names)
-    except DictionaryLoaderException as e:
-        raise InvalidConfigurationError(unicode(e))
-    engine.get_dictionary().set_dicts(dicts)
-
-    engine.set_log_file_name(config)
-    engine.enable_stroke_logging(config.get_enable_stroke_logging())
-    engine.enable_translation_logging(config.get_enable_translation_logging())
-    engine.set_space_placement(config.get_space_placement())
-    engine.set_starting_stroke_state(capitalize=config.get_start_capitalized(),
-                                     attach=config.get_start_attached())
-
+    update_engine(engine, None, config)
     engine.set_is_running(config.get_auto_start())

 def reset_machine(engine, config):
     """Set the machine on the engine based on config."""
-    machine_type = config.get_machine_type()
-    machine_options = config.get_machine_specific_options(machine_type)
-    try:
-        instance = machine_registry.get(machine_type)(machine_options)
-    except NoSuchMachineException as e:
-        raise InvalidConfigurationError(unicode(e))
-    engine.set_machine(instance)
-
-def update_engine(engine, old, new):
+    update_engine(engine, config, config, reset_machine=True)
+
+def update_engine(engine, old, new, reset_machine=False):
     """Modify a StenoEngine using a before and after config object.

     Using the before and after allows this function to not make unnecessary
     changes.
     """
+    if old is None:
+        # Fake config that will return None for each
+        # option, hence triggering an update.
+        class NoneConfig(object):
+            def __getattr__(self, name):
+                return lambda *args: None
+        old = NoneConfig()
+
     machine_type = new.get_machine_type()
     machine_options = new.get_machine_specific_options(machine_type)
-    if (old.get_machine_type() != machine_type or
+    if (reset_machine or
+        old.get_machine_type() != machine_type or
         old.get_machine_specific_options(machine_type) != machine_options):
         try:
             machine_class = machine_registry.get(machine_type)
```